Source/JavaScriptCore/b3/B3LowerToAir.cpp

```diff
@@ -3384,10 +3384,10 @@ class LowerToAir {
                 uint64_t mask = maskValue->asInt();
                 if (!mask || mask & (mask + 1))
                     return false;
-                uint64_t maskBitCount = WTF::bitCount(mask);
-                uint64_t highWidth = highWidthValue->asInt();
-                uint64_t lowWidth = lowWidthValue->asInt();
-                uint64_t datasize = opcode == ExtractRegister32 ? 32 : 64;
+                int64_t maskBitCount = WTF::bitCount(mask);
+                int64_t highWidth = highWidthValue->asInt();
+                int64_t lowWidth = lowWidthValue->asInt();
+                int64_t datasize = opcode == ExtractRegister32 ? 32 : 64;
                 // Note that when `lowWidth == datasize` we cannot turn it to `MOV Rd Rn` since
```

Comment:
These values cannot be negative. So, we should use unsigned types here. In order to get grid of unexpected warning, we should explicitly cast them to `uint64_t`.

Reply:
> These values cannot be negative.

I mean `highWidth` and `lowWidth` can have negative values. The intent here is find the pattern that `0 <= lowWidth < datasize` and `0 < highWidth <= datasize` and `lowWidth  + highWidth == datasize`.

Reply:
The warning is not on these lines.
If you want to keep the `uint64_t` variables, the assert needs a cast :
```
ASSERT(0 <= static_cast<int64_t>(lowWidth) && lowWidth < datasize);
```

Reply:
I see, let's keep the unsigned types and change the assertion to
```
ASSERT(lowWidth < datasize);
```
since `uint64_t lowWidth` must be `>= 0`.

Reply:
OK, let's go back to the old patch then.